### ISSUE TITLE
Trailing slash problem if request: Rest in parameters

### ISSUE THREAD

**Issue by boumboum** [tiangolo/fastapi#3407]
Given that def:
```
@app.get("/allcounters", response_class=HTMLResponse)
async def allcounters(request: Request):
    return templates.TemplateResponse('allcounters.html', context={'request': request, 'cards': counter_data})
```

and the URL looks like that:
`http://localhost:8000/allcounters`

Than it causes this error:
`{"detail":[{"loc":["path","id"],"msg":"value is not a valid integer","type":"type_error.integer"}]}`

But if I change the decorator to that:
`@app.get("/allcounters/", response_class=HTMLResponse)`

and the URL call like that:
`http://localhost:8000/allcounters/`

it works.
I suspect the `request: Request` parameter in my def causing the issue.
The expectation of course would be to get it working without leading slash.
Any clue?

**Comment by Kludex** [comment]
You're probably calling another endpoint? https://fastapi.tiangolo.com/tutorial/path-params/#order-matters

**Comment by boumboum** [comment]
Not really. In between I've tested it with other endpoints and I've discovered that if I'm not making a pure API call (which works fine) but rendering a web page, this issue systematically appears, regardless of the parameters.

**Comment by boumboum** [comment]
That was the problem. Thanks for the help. The issue can be closed.
